### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Xaenalt**

### CODE REVIEW

**Comment by Xaenalt** [comment]
Need to properly rebase this, since I ended up writing this on the branch I was working on other stuff on

/hold

File: test/smoke-test.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+mv flan-t5-small-caikit /opt/models/
+
+CAIKIT_LOG_FILE=$(mktemp)
+cd /caikit
+./start-serving.sh &>$CAIKIT_LOG_FILE &
+# timeout --signal=SIGINT 30 bash -c "tail -f -n0 $CAIKIT_LOG_FILE | grep 'Caikit Runtime is serving on port'"
+timeout --signal=SIGINT 30 bash -c "grep -m1 'Caikit Runtime is serving on port' <(tail -f $CAIKIT_LOG_FILE)"
+
+grpcurl -plaintext -d '{"text": "At what temperature does liquid Nitrogen boil?"}' -H "mm-model-id: flan-t5-small-caikit" localhost:8085 caikit.runtime.Nlp.NlpService/TextGenerationTaskPredict
```
Comment (danielezonca) [inline test/smoke-test.sh:17]: What about adding a check on return code or return value that is not empty?

File: Makefile [hunk 2]
```diff
@@ -8,3 +8,13 @@ default:
 
 refresh-piplock-files:
 	cd caikit && pipenv lock
+
+.PHONY: docker-test
+
+docker-test: default
+	podman run -it --rm \
+	-e DTYPE_STR=float32 \
```
Comment (danielezonca) [inline Makefile:16]: Is it possible to move this conf to the template? My understanding is that it is necessary to run on CPU. Correct?

File: Makefile [hunk 3]
```diff
@@ -8,3 +8,13 @@ default:
...
+	--name caikit-tgis-serving-test-$$(git rev-parse --short HEAD) \
+	--volume $$(pwd)/test:/tmp/test:z --volume $$(pwd)/utils:/tmp/utils:z \
+	caikit-tgis-serving:$$(git rev-parse --short HEAD) \
```
Comment (danielezonca) [inline Makefile:19]: As far as I see this script requires the image to be already build, does it make sense to build the image as part of this script to have everything in a single command?

File: Makefile (hunk 2 above)
Comment (Xaenalt) [reply Makefile:16]: It is, but we don't want to set it universally, since GPU workloads may need to set it differently

File: Makefile (hunk 3 above)
Comment (Xaenalt) [reply Makefile:19]: Yes, build is a dependency of this target https://github.com/opendatahub-io/caikit-tgis-serving/pull/34/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R14

File: test/smoke-test.sh (hunk 1 above)
Comment (Xaenalt) [reply test/smoke-test.sh:17]: https://github.com/opendatahub-io/caikit-tgis-serving/pull/34/files#diff-ddd7611de37976c7d21b1c8eec98863960dc45a5b6c2ca4b971d06939192be7eR18 is the return code from it

Comment (danielezonca) [reply test/smoke-test.sh:17]: Ok now I understand how it works, the script returns the return code of the grpcurl code 👍

File: Makefile (hunk 2 above)
Comment (danielezonca) [reply Makefile:16]: Ok, do you suggest to document that this option is necessary to run on CPU?
Here is a bit hidden in the script

Comment (Xaenalt) [reply Makefile:16]: It's not needed forever, need is only based on https://github.com/IBM/text-generation-inference/issues/7

Comment (Xaenalt) [reply Makefile:16]: By the time we're ready for release, should hopefully no longer be necessary

**Pull request merged**